superset/views/annotations.py

```diff
@@ -14,117 +14,27 @@
 # KIND, either express or implied.  See the License for the
 # specific language governing permissions and limitations
 # under the License.
-from typing import Any, Dict
-
-from flask_appbuilder import CompactCRUDMixin
+from flask_appbuilder import permission_name
 from flask_appbuilder.api import expose
-from flask_appbuilder.models.sqla.interface import SQLAInterface
 from flask_appbuilder.security.decorators import has_access
-from flask_babel import lazy_gettext as _
-from wtforms.validators import StopValidation
 
-from superset.constants import MODEL_VIEW_RW_METHOD_PERMISSION_MAP, RouteMethod
-from superset.models.annotations import Annotation, AnnotationLayer
 from superset.superset_typing import FlaskResponse
-from superset.views.base import SupersetModelView
-
-
-class StartEndDttmValidator:  # pylint: disable=too-few-public-methods
-    """
-    Validates dttm fields.
-    """
 
-    def __call__(self, form: Dict[str, Any], field: Any) -> None:
-        if not form["start_dttm"].data and not form["end_dttm"].data:
-            raise StopValidation(_("annotation start time or end time is required."))
-        if (
-            form["end_dttm"].data
-            and form["start_dttm"].data
-            and form["end_dttm"].data < form["start_dttm"].data
-        ):
-            raise StopValidation(
-                _("Annotation end time must be no earlier than start time.")
-            )
```

Comment:
Note: I verified that the CRUD modal takes care of this validation in the frontend and that this validation is done in the command.

Bycatch while checking the related code: This is trying to append the `Type` to `exceptions`, should be an instance of `AnnotationDatesValidationError`

https://github.com/apache/superset/blob/ac0ff78616119bd5d8bebbb2781e0ef3486c4d19/superset/annotation_layers/annotations/commands/create.py#L73

Why on earth is MyPy not picking this up?

Reply:
very nice catch! fixing